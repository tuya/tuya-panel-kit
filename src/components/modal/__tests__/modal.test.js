import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';
import Modal from '../index';

describe('Modal Component', () => {
  it('basic render', () => {
    const component = renderer.create(<Modal visible={false} onMaskPress={jest.fn()} />);
    expect(component).toMatchSnapshot();
    component.update(<Modal visible={true} onMaskPress={jest.fn()} />);
    component.unmount();
    expect(component).toMatchSnapshot();
  });

  it('render with visible', () => {
    const component = renderer.create(<Modal visible={true} onMaskPress={jest.fn()} />);
    expect(component).toMatchSnapshot();
    component.update(<Modal visible={false} onMaskPress={jest.fn()} />);
    component.unmount();
    expect(component).toMatchSnapshot();
  });

  it('render with visible no changed', () => {
    const component = renderer.create(<Modal visible={true} onMaskPress={jest.fn()} />);
    expect(component).toMatchSnapshot();
    component.update(<Modal visible={true} onMaskPress={jest.fn()} />);
    component.unmount();
    expect(component).toMatchSnapshot();
  });

  it('render with modal.render', () => {
    const component = renderer.create(
      Modal.render(
        <View>
          <Text style={{ color: '#333', textAlign: 'center', backgroundColor: '#fff' }}>
            {'tuya'}
          </Text>
          <TouchableOpacity
            onPress={jest.fn()}
            style={{ height: 44, backgroundColor: '#fff', justifyContent: 'center', marginTop: 20 }}
          >
            <Text style={{ color: '#333', textAlign: 'center', backgroundColor: '#fff' }}>
              {'changeText'}
            </Text>
          </TouchableOpacity>
        </View>
      )
    );
    expect(component).toMatchSnapshot();
  });

  it('render with modal.close', () => {
    const component = renderer.create(Modal.close());
    expect(component).toMatchSnapshot();
  });
});
