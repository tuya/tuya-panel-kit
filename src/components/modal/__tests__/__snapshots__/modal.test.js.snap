// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component basic render 1`] = `null`;

exports[`Modal Component basic render 2`] = `null`;

exports[`Modal Component render with modal.close 1`] = `null`;

exports[`Modal Component render with modal.render 1`] = `null`;

exports[`Modal Component render with visible 1`] = `null`;

exports[`Modal Component render with visible 2`] = `null`;

exports[`Modal Component render with visible no changed 1`] = `null`;

exports[`Modal Component render with visible no changed 2`] = `null`;
